nextcloudappstore/core/views.py

```diff
@@ -29,8 +32,33 @@ class AppDetailView(DetailView):
     slug_field = 'id'
     slug_url_kwarg = 'id'
 
+    def post(self, request, id):
+        user = request.user
+        form = AppRatingForm(request.POST)
+        # there is no way that a rating can be invalid by default
+        if form.is_valid() and user.is_authenticated():
+            app = App.objects.get(id=id)
+            rating, created = AppRating.objects.get_or_create(user=user,
+                                                              app=app)
+            rating.rating = form.cleaned_data['rating']
+            rating.set_current_language(request.LANGUAGE_CODE)
+            rating.comment = form.cleaned_data['comment']  # TODO: locale
+            rating.save()
+        return redirect('app-detail', id=id)
+
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
+        try:
+            app_rating = AppRating.objects.get(user=self.request.user,
+                                               app=context['app'])
+            context['rating_form'] = AppRatingForm(initial={
+                'rating': app_rating.rating,
+                'comment': app_rating.comment
+            })
+            context['user_rated_app'] = True
```

Comment:
"user_rated_app" sounds like noun. I'd name this something like "user_has_rated" or "user_has_rated_app", although "app" might be a bit redundant since we are in the app detail view.